.cirrus.yml

```diff
@@ -78,14 +82,19 @@ functional_task:
       # Installing it from Hashicorp directly
     - wget -O- https://apt.releases.hashicorp.com/gpg | gpg --dearmor -o /usr/share/keyrings/hashicorp-archive-keyring.gpg
     - echo "deb [signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] https://apt.releases.hashicorp.com $(lsb_release -cs) main" | tee /etc/apt/sources.list.d/hashicorp.list
-    - DEBIAN_FRONTEND=noninteractive apt-get -y update && DEBIAN_FRONTEND=noninteractiven apt-get -y install vagrant ruby-libvirt qemu-kvm virt-manager libvirt-daemon-system virtinst libvirt-clients bridge-utils pkg-config libxslt-dev libxml2-dev libvirt-dev zlib1g-dev ruby-dev gcc make
+    - DEBIAN_FRONTEND=noninteractive apt-get -y update && DEBIAN_FRONTEND=noninteractiven apt-get -y install vagrant ruby-libvirt qemu-kvm virt-manager libvirt-daemon-system virtinst libvirt-clients bridge-utils pkg-config libxslt-dev libxml2-dev libvirt-dev zlib1g-dev ruby-dev gcc make ruby-nokogiri
     - vagrant plugin install vagrant-libvirt
     - systemctl enable --now libvirtd
 
   # Download, provision and cache test image(s).
   vagrant_cache: &vagrant_cache
-    fingerprint_script: uname -s ; cat Vagrantfile
-    folder: /root/.vagrant.d
+    fingerprint_script: |
+      # for rawhide invalidate once a week
+      if [ x${DISTRO} = xrawhide ]; then
+          date -d 'next monday - 1 week' +%Y%m%d
+      fi
+      echo ${DISTRO}; uname -s ; cat Vagrantfile
```

Comment:
> As we're using a `matrix` instruction, the task is actually per-distro. Using `echo ${DISTRO}` is only useful is the cache is global to all tasks, otherwise it won't be needed. Do you know if that's the case?
> 
> Also, not that it's linked to your modifications, but I'm wondering why we use `uname -s` as part of the cache key. It'll always return the same value.

I'm not sure about the matrix assumption.
As far as I can tell looking at the output of old runs is that usually we always have a hit (Vagrantfile + uname -s) but only one image was really part of the cache (the one uploaded after the first cache miss).

I guess `uname -s` exists in the first place because of:
```Cache artifacts are shared between tasks, so two caches with the same name on e.g. Linux containers and macOS VMs will share the same set of files. This may introduce binary incompatibility between caches. To avoid that, add echo $CIRRUS_OS into fingerprint_script or use $CIRRUS_OS in fingerprint_key, which will distinguish caches based on OS.```

Reply:
> `Cache artifacts are shared between tasks, so two caches with the same name [...] will share the same set of files`. So that means our cache will be reused across all the distro tasks. The issue I'm seeing now is that by adding `echo ${DISTRO}` the cache will be invalidated each time (more or less, there could be rare cases where it's not). So while the cache will be used for tasks, it gets re-uploaded each time.
> 

this is not what I observed running the same task manually multiple times. 
Just to clarify what I think it happens.
Old way:
First task that run to completion (let's say f38) populates the globally shared
`*/hash/files` 

Now the cache is immutable and only changes when the key changes (basically Vagrantfile).
F38 will always be actually cached, whereas the remaining will always be downloaded. This was observed in old runs logs (might need another check).

Now the approach is

pr/hashX/files
pr/hashY/files
pr/hashZ/files

Where X and Y differ because of ${DISTRO}, while Z is ${DISTRO} + the first day of the current week.
When f38 runs, the key matches hashX and the related files get downloaded. Same for Y and Z (rawhide).

> We're trying to have the same cache for different _incompatible_ tasks. I think we should stop having a common cache and split it per-distro. We could define it in the `matrix` and use a template if we want.
> 

This was the idea, see above.

> Also given your explanations, `uname -s` is not required for us and as we're Linux dependent won't probably be.

yep, I agree.
Please note that `fedora.vm.box_check_update` is a leftover (it's useless and harmless in our current cases) from previous tests.

Reply:
you can verify if the cache is working in the run up logs. If the image gets downloaded, then it's not working, otherwise, it is.